### PR TITLE
Fix modules in Special Term do not show up in Today page

### PR DESCRIPTION
**Pull request by xinyuwang-nus**
## Context

https://github.com/nusmodifications/nusmods/issues/3757
https://github.com/nusmodifications/nusmods/issues/3830

### CODE REVIEW

**Review by ravern — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing this!

I've added https://github.com/nusmodifications/nusmods/issues/3757#issuecomment-2383138799 after doing a deep dive into the issue.

I'm good to merge this once the linting errors are fixed, since this solves the issue for now. You can find the linting errors by clicking the "Details" link in the failing "build_and_test" workflow, then navigating to the "Tests" tab.

Here's a direct link to that page: https://app.circleci.com/pipelines/github/nusmodifications/nusmods/14863/workflows/fc3ce863-17e5-49a2-84e9-2d99e12ab411/jobs/89894/tests

**Comment by xinyuwang-nus** [comment]
> Sorry for the delay in reviewing this!
> 
> I've added [#3757 (comment)](https://github.com/nusmodifications/nusmods/issues/3757#issuecomment-2383138799) after doing a deep dive into the issue.
> 
> I'm good to merge this once the linting errors are fixed, since this solves the issue for now. You can find the linting errors by clicking the "Details" link in the failing "build_and_test" workflow, then navigating to the "Tests" tab.
> 
> Here's a direct link to that page: https://app.circleci.com/pipelines/github/nusmodifications/nusmods/14863/workflows/fc3ce863-17e5-49a2-84e9-2d99e12ab411/jobs/89894/tests

File: website/src/views/today/TodayContainer/TodayContainer.tsx
[the remark this answers is not in the document]
Comment (ravern) [reply website/src/views/today/TodayContainer/TodayContainer.tsx:1]: Looks good, and thank you for adding these comprehensive tests! I think the inconsistencies are small and can be fixed in a follow-up PR, I'm more interested in merging this one which fixes the bigger bug.

[the remark this answers is not in the document]
Comment (xinyuwang-nus) [reply website/src/views/today/TodayContainer/TodayContainer.tsx:1]: sounds good

**Review by ravern — APPROVED** [review]
LGTM, thank you for addressing this issue! Definitely not a trivial one to solve.

**Pull request merged**